+nosnoc/+discrete_time_problem/Heaviside.m

```diff
@@ -363,28 +399,49 @@ function generate_direct_transcription_constraints(obj)
                     obj.f = obj.f + dcs.f_q_fun(x_ijk, z_ijk, alpha_ijk, lambda_n_ijk, lambda_n_ijk, u_i, v_global, p);
                 end
 
-                % Clock <Constraints
+                % Clock Constraints
                 % TODO(@anton) HERE BE DRAGONS. This is by far the worst part of current nosnoc as it requires the discrete problem
                 %              to understand something about the time-freezing reformulation which is ugly.
-                if obj.opts.use_fesd && opts.equidistant_control_grid
-                    if opts.time_optimal_problem
-                        if opts.use_speed_of_time_variables
-                            obj.g.equidistant_control_grid(ii) = {[sum_h - opts.h;s_sot*sum_h - obj.w.T_final()/opts.N_stages]};
-                        else
-                            obj.g.equidistant_control_grid(ii) = {sum_h - obj.w.T_final()/opts.N_stages};
-                        end
+                % handle numerical time
+                if opts.use_fesd && opts.equidistant_control_grid
```

Comment:
if we have time freezing, the equidistant control grid in numerical time has not really a meaning, but adds more constraints. this might slow down convergence? 
hence, does it make sense to always turn equidistant_control_grid off as soon as we have time freezing on?

Reply:
On the contrary including the numerical time constraints improves convergence in my experience. As discussed there isn't necessarily a good rigorous reason as to why this should be, but for now we keep it I think.

Reply:
If it improves convergence, then it has it justification. 
It restrictis additionaly the feasible set, might be just a non-obvious guiding constraint.